### PR TITLE
Added sign up page with functionality

### PR DESCRIPTION
**Pull request by Shah-Arsalan**
**Added sign up functionality **

- The user can now access the signup page and create a new account 
- Creating a new account will redirect the user to the homepage and the user can now add products to cart and wishlist 
Screenshot (216)

### CODE REVIEW

File: src/Contexts/auth-context.jsx [hunk 1]
```diff
@@ -11,12 +11,39 @@ const AuthProvider = ({ children }) => {
   const [token, setToken] = useState(localStorageItems?.userToken);
   const [user, setUser] = useState(localStorageItems?.activeUser);
   const loginCall = async (email, password) => {
+    console.log(email, password);
```
Comment (manish591) [inline src/Contexts/auth-context.jsx]: Remove Console if you don't need it

File: src/Contexts/auth-context.jsx [hunk 2]
```diff
@@ -11,12 +11,39 @@ const AuthProvider = ({ children }) => {
...
     try {
       const response = await axios.post("/api/auth/login", {
         email,
         password,
       });
 
+      if (response.status === 200 || response.status === 201) {
+        console.log("success", email, password);
```
Comment (manish591) [inline src/Contexts/auth-context.jsx]: Remove console.

File: src/Contexts/auth-context.jsx [hunk 3]
```diff
@@ -11,12 +11,39 @@ const AuthProvider = ({ children }) => {
...
+          JSON.stringify({
+            userToken: response.data.encodedToken,
+            activeUser: response.data.foundUser,
+          })
+        );
+        setToken(response.data.encodedToken);
+        setUser(response.data.foundUser);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (manish591) [inline src/Contexts/auth-context.jsx]: For error messages, you can use `console.error`, but you can stick with this as well.

File: src/WebPages/loginPage/Loginpage.css [hunk 4]
```diff
@@ -47,3 +47,11 @@ input[type="password"] {
 input[type="email"] {
```
Comment (manish591) [inline src/WebPages/loginPage/Loginpage.css:47]: Use a className to target the input element.

File: src/WebPages/signup/Signup.jsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
+import { useState } from "react";
+import { useEffect } from "react";
```
Comment (manish591) [inline src/WebPages/signup/Signup.jsx:2]: You can just write one rule like this, `import { useState, useEffect } from react`. But, if you like this way you can for it.

File: src/WebPages/signup/Signup.jsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+  return (
+    <>
+      <div className="signup-page">
+        <div className="login-container">
+          <div className="login-inputs">
+            <h1>Sign Up</h1>
+            <div className="input">
+              <label>Email</label>
+              <input
```
Comment (manish591) [inline src/WebPages/signup/Signup.jsx]: Your label and input element are not connected. There is no point in using labels without htmlFor attribute.
Why you should do this? 
It increases the tap target.

File: src/WebPages/loginPage/LoginPage.jsx [hunk 7]
```diff
@@ -8,18 +8,23 @@ const LoginPage = () => {
...
+  };
 
   useEffect(() => {
     if (token) {
       setTimeout(() => {
         navigate("/");
       }, 1000);
     }
-  }, [token]);
+  }, [token, loginDetails]);
```
Comment (anuj-kumary) [inline src/WebPages/loginPage/LoginPage.jsx]: No need to give loginDetails as a dependency here

**Review by anuj-kumary — APPROVED** [review]
Your signup functionality working good. Readable code and good folder structure.

**Review by manish591 — CHANGES_REQUESTED** [review]

**Review by Logan1x — COMMENTED** [review]
The unnecessary console's are there but otherwise all okay.

File: src/WebPages/signup/Signup.jsx (hunk 6 above)
Comment (Shah-Arsalan) [reply src/WebPages/signup/Signup.jsx]: Thanks for your review manish . I'll do the changes

**Review by Logan1x — APPROVED** [review]

**Pull request merged**